components/measurement/nettests/facebook_messenger.js

```diff
@@ -33,24 +34,46 @@ export const FacebookMessengerDetails = ({ measurement, render }) => {
   "facebook_stun_dns_consistent": true,
   "facebook_stun_reachable": null,
   */
+
+  let summaryText = '|'
+  if (!isWorking) {
+    if (tcpBlocking) {
+      summaryText += intl.formatMessage({id: 'Measurement.Details.SummaryText.FacebookMessenger.TCPFailure'})
+    } else {
+      summaryText += intl.formatMessage({id: 'Measurement.Details.SummaryText.FacebookMessenger.TCPSuccess'})
+    }
+
+    if (dnsBlocking) {
+      summaryText += intl.formatMessage({id: 'Measurement.Details.SummaryText.FacebookMessenger.DNSFailure'})
+    } else {
+      summaryText += intl.formatMessage({id: 'Measurement.Details.SummaryText.FacebookMessenger.DNSSuccess'})
```

Comment:
It seems like you are mixing in here the use of the `summaryText` variable for both `FormattedMessage` `id` key values and also the actual translated string.

Does this work as expected?

Reply:
Turns out it is a bug, yet works as expected. It was certainly not intended this way. `react-intl` fails to find the key and prints the key itself as the message.

I have [modified `SummaryText`](https://github.com/ooni/explorer/pull/146/files#diff-77ee24bf0b058edd4a84b18feaddef79) to accept either a `function`, `string`, or `any` in the `content` prop and render them accordingly. The problem with `string` type is that it cannot distinguish between the `react-intl` key and an actual string to be rendered (in this case already formatted).

One fix is sending the key as an object to the `content` prop e.g `content={{ key: 'Measurement.Test.SummaryText.Reason'}}` with additional values if needed. And handle it the way `typeof content === 'string'` is handled right now. And render strings as is.

I feel the measurement details section in general and the SummaryText component in particular have gone through many iterations and I haven't been able to apply the patterns consistently across `components/measurement/nettests/*.js` files. I wonder if I should refactor it based on what we know now about what we want.

Reply:
I understand. Maybe file an issue related to this part of refactoring so that we don't forget about it and we can proceed with merging this as is for the time being.

Reply:
https://github.com/ooni/explorer/issues/156